tests/lib/rules/prefer-user-event.test.ts

```diff
@@ -0,0 +1,123 @@
+              import { fireEvent } from '${libraryModule}'
+              const node = document.createElement(elementType)
+              fireEvent.${fireEventMethod}(foo)
+            `,
+            options: [{ allowedMethods: [fireEventMethod] }]
+          }))
+      )
+    , []),
+    ...LIBRARY_MODULES.reduce((acc: ValidTestCase<Options>[], libraryModule) => 
+      acc.concat(
+        Object
+          .keys(MappingToUserEvent)
+          .map((fireEventMethod) => ({
+            code: `
+              import { fireEvent as fireEventAliased } from '${libraryModule}'
+              const node = document.createElement(elementType)
+              fireEventAliased.${fireEventMethod}(foo)
+            `,
+            options: [{ allowedMethods: [fireEventMethod] }]
+          }))
+      )
+    , []),
+    ...LIBRARY_MODULES.reduce((acc: ValidTestCase<Options>[], libraryModule) => 
+      acc.concat(
+        Object
+          .keys(MappingToUserEvent)
+          .map((fireEventMethod) => ({
+            code: `
+              import * as dom from '${libraryModule}'
+              dom.fireEvent.${fireEventMethod}(foo)
+            `,
+            options: [{ allowedMethods: [fireEventMethod] }]
+          }))
+      )
+    , []),
+    ...LIBRARY_MODULES.map((libraryModule) => ({
+      // imported fireEvent and not used, 
+      code: `
+        import { fireEvent } from '${libraryModule}'
+        import * as foo from 'someModule'
+        foo.baz()
+      `
+    })),
+    ...LIBRARY_MODULES.map((libraryModule) => ({
+      // imported dom, but not using fireEvent
+      code: `
+        import * as dom from '${libraryModule}'
+        const button = dom.screen.getByRole('button')
+        const foo = dom.screen.container.querySelector('baz')
+      `
+    })),
+    ...LIBRARY_MODULES.map((libraryModule) => ({
+      code: `
+        import { fireEvent as aliasedFireEvent } from '${libraryModule}'
+        function fireEvent() {
+          console.log('foo')
+        }
+        fireEvent()
+    `
+    })),
+  ],
+  invalid: [
+    ...LIBRARY_MODULES.reduce((acc: InvalidTestCase<MessageIds, Options>[], libraryModule) => 
+      acc.concat(
+        Object.keys(MappingToUserEvent).map((fireEventMethod: string) => ({
+          code: `
+            import { fireEvent } from '${libraryModule}'
+            const node = document.createElement(elementType)
+            fireEvent.${fireEventMethod}(foo)
+          `,
+          errors: [{
+            messageId: 'preferUserEvent'
+          }]
+        }))
+      ),
+    []),
+    ...LIBRARY_MODULES.reduce((acc: InvalidTestCase<MessageIds, Options>[], libraryModule) => 
+      acc.concat(
+        Object.keys(MappingToUserEvent).map((fireEventMethod: string) => ({
+          code: `
+            import * as dom from '${libraryModule}'
+            dom.fireEvent.${fireEventMethod}(foo)
+          `,
+          errors: [{ messageId: 'preferUserEvent' }]
+        }))
+      ),
+    [])
+  ]
+});
```

Comment:
There's a bunch of repetitive mapping and reducing here, maybe we could extract helper functions

Reply:
done - definitely the scenarios are a bit more readable now. Thanks!